src/main/java/com/academia/library/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+package com.academia.library.service.impl;
+
+import com.academia.library.dto.OrderRequest;
+import com.academia.library.dto.OrderResponse;
+import com.academia.library.exception.OrderNotFoundException;
+import com.academia.library.exception.UserNotFoundException;
+import com.academia.library.mapper.OrderMapper;
+import com.academia.library.model.Order;
+import com.academia.library.model.OrderStatus;
+import com.academia.library.model.User;
+import com.academia.library.repository.OrderDetailsRepository;
+import com.academia.library.repository.OrderRepository;
+import com.academia.library.repository.UserRepository;
+import com.academia.library.service.OrderService;
+import com.academia.library.validator.OrderValidator;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderDetailsRepository orderDetailsRepository;
+    private final OrderMapper orderMapper;
+    private final OrderValidator orderValidator;
+    private final UserRepository userRepository;
+
```

Comment:
Please move methods like this:

private final OrderRepository orderRepository;
private final OrderDetailsRepository orderDetailsRepository;
private final UserRepository userRepository;
private final OrderMapper orderMapper;
private final OrderValidator orderValidator;

Reply:
Ok, thank